iina/VideoView.swift

```diff
@@ -285,8 +285,12 @@ class VideoView: NSView {
 
     if videoLayer.colorspace != nil {
       Logger.log("Nilling out colorspace", subsystem: hdrSubsystem)
-      videoLayer.colorspace = nil;
       videoLayer.wantsExtendedDynamicRangeContent = false
+      // It seems that once there is a layer with colorspace set to something extended linear,
+      // then setting colorspace to nil will cause falling back to ExtendedLinearsRGB
+      videoLayer.colorspace = CGColorSpaceCreateDeviceRGB()
```

Comment:
colorspace is nil by default. Disabling HDR mode should revert all changes to the default value as HDR was never enabled. In addition, since we are testing `videoLayer.colorspace != nil` in [Line 286](https://github.com/iina/iina/pull/4174/files#diff-0da7dedcb527d7cb82822500e08f5a55ee5fb2a7fb99b2bf039dbad1b6787546L286), setting `videoLayer.colorspace` to non-nil makes the resetting code be executed every time when setICCProfile being called

Reply:
The problem was, as I mentioned in the commit, once `videoLayer.colorspace` was set to `CGColorspace.extendedLinear*` before, then setting it to `nil` would *not* clear it, instead, it would fallback to `kCGColorspaceExtendedLinearSRGB` and completely throw off SDR color. I'm not sure whether it is the intended behavior.

Setting it to `CGColorspaceCreateDeviceSRGB()` should also turn off color management on the buffer like `nil`. I do agree condition should be updated, though.

Attached is the screenshot trying to turn off HDR without the last commit d9438770d7c625b37bcc7e37bbf3adf5f1ac165b. 
![Linear](https://user-images.githubusercontent.com/30361859/211966944-9a814581-614b-4372-a446-0e8edbbb96ea.jpg)

Reply:
It turns out that it is `kCGLPFAColorFloat, kCGLPFAColorSize, CGLPixelFormatAttribute(64),` that makes nil fallback to `kCGColorSpaceExtendedLinearSRGB`. You may remove these code and try again

Reply:
`static let SRGB = CGColorSpaceCreateDeviceRGB()`

`videoLayer.colorspace = VideoView.SRGB`

`if videoLayer.colorspace != VideoView.SRGB { ...`

Reply:
> It turns out that it is `kCGLPFAColorFloat, kCGLPFAColorSize, CGLPixelFormatAttribute(64),` that makes nil fallback to `kCGColorSpaceExtendedLinearSRGB`. You may remove these code and try again

The problem is output of 10-bit color depth videos would be incorrect without them. These ensure pixel format of OpenGL buffer be 16-bit float-point instead of default 8-bit.

The following test pattern `Mehanik HDR10 test patterns/03. Grayscale/04. 10bit test/01. test_10-bit_23.976.mp4` from https://www.avsforum.com/threads/hdr10-test-patterns-set.2943380/ will demonstrate the problem. Unfortunately, because all screenshots I could take would be clamped to 8-bit color depth, I cannot directly provide a picture here.
Direct download link: [test-10bit.mp4](https://user-images.githubusercontent.com/30361859/212476493-489a114e-06d5-43d1-a2e4-43f9063db855.mp4)

Reply:
I have a lot of HDR test videos.

I meant that it was not `once videoLayer.colorspace was set to CGColorspace.extendedLinear* before` caused the problem

Reply:
You're right indeed, the comment and problematic commit message were removed.